### DISCUSSION TITLE
Increase chunk size

### DISCUSSION THREAD

**Discussion by hakavlad** [samuel-lucas6/Kryptor#72] [Ideas]
>The plaintext file is then read and encrypted in 16 KiB chunks. Once the end of the file has been reached, whatever was previously read into the plaintext buffer is encrypted in chunks as padding until the padded length is reached. The final chunk can be less than 16 KiB.

Why did you choose 16 KiB? Encryption in 64 (like `age`) or 128 KiB chunks is faster.

**Comment by samuel-lucas6** [comment]
It was primarily based on [Adam Langley's suggestion](https://www.imperialviolet.org/2014/06/27/streamingencryption.html)/the TLS record size. From memory, I think it was also influenced by the fact that the final chunk wasn't truncated in some older versions and possibly other reasons too.
